src/main/java/ru/hogwarts/school/model/Faculty.java

```diff
@@ -1,15 +1,34 @@
 package ru.hogwarts.school.model;
 
+import nonapi.io.github.classgraph.json.Id;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.ManyToMany;
 import java.util.Objects;
+import java.util.Set;
 
+@Entity
 public class Faculty {
+    @javax.persistence.Id
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
     private String color;
-    public Faculty(Long id,String name,String color){
-        this.id=id;
-        this.name=name;
-        this.color=color;
+    @ManyToMany(mappedBy = "faculty")
```

Comment:
Тут нужно поменять аннотации на такие:
```suggestion
    @OneToMany(mappedBy = "faculty")
    @JsonIgnore
```